### PR TITLE
Migrate the various map impls in the CLDR transformer to LiteMap

### PR DESCRIPTION
**Pull request by sffc**
More progress on #568

Depends on #1340 

We had several different map implementations in different places, including:

- tuple_vec_map
- HashMap
- BTreeMap

I have replaced all of them with LiteMap.  I chose LiteMap because:

1. It is compatible with tuple_vec_map, which was the most common map
2. We can add methods to litemap, like #1340, to make it work better for our use case
3. This is a good use of LiteMap to prevent it from bitrotting

CC @samchen61661: The only remaining uses of `tuple_vec_map` are in the LocaleCanonicalizer transformer.  I hope you can remove those as part of your work to transition that class to LiteMap.  I could have done it here, but I didn't want to conflict with your work.

### CODE REVIEW

File: utils/litemap/src/map.rs [hunk 1]
```diff
@@ -141,6 +141,42 @@ impl<K: Ord, V> LiteMap<K, V> {
         }
     }
 
+    /// Get the lowest-rank key/value pair from the `LiteMap`, if it exists.
```
Comment (sffc) [inline utils/litemap/src/map.rs:144]: NOTE: The changes in this file should be reviewed in a standalone PR, #1340.

**Review by Manishearth — APPROVED** [review]
